core/desugarMatching.ml

```diff
@@ -43,12 +43,12 @@ let pattern_matching_sugar_guard pos =
   if not (Settings.get pattern_matching_sugar)
   then raise (pattern_matching_sugar_disabled pos)
 
-let nullary_guard tuple pos =
+let nullary_guard pss pos =
   let nullary_error pos =
     Errors.desugaring_error ~pos:pos ~stage:Errors.DesugarMatching ~message:"Can't match over nullary function"
   in
-  match tuple with
-    | [] -> raise (nullary_error pos)
+  match pss with
+    | [[]] -> raise (nullary_error pos)
     | _ -> ()
```

Comment:
We also need to check whether the function is curried, as we decided not to support this just yet. The following should be rejected
```
links> fun f(x)(y) switch { case x -> x };
f = fun : (a) -> (c) ~> (a, c)
links> f(2)(4);
(2, 4) : (Int, Int)
```